keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -2788,79 +2874,76 @@ previous answers based on experience in the field.
 
 ###### Will enabling / using this feature result in any new API calls?
 
-<!--
-Describe them, providing:
-  - API call type (e.g. PATCH pods)
-  - estimated throughput
-  - originating component(s) (e.g. Kubelet, Feature-X-controller)
-Focusing mostly on:
-  - components listing and/or watching resources they didn't before
-  - API calls that may be triggered by changes of some Kubernetes resources
-    (e.g. update of object X triggers new updates of object Y)
-  - periodic API calls to reconcile state (e.g. periodic fetching state,
-    heartbeats, leader election, etc.)
--->
+For Pods not using ResourceClaims, not much changes. The following components
+need to watch additional resources:
+- kube-controller-manager: ResourceClaimTemplate
+- kube-scheduler: ResourceClaim, DeviceClass, ResourceSlice
+- kubelet: ResourceClaim
+
+If the feature isn't used, those watches will not cause much overhead.
+
+If the feature is used, kube-scheduler needs to update ResourceClaim during Pod
+scheduling kube-scheduler. kube-controller-manager creates it (optional, only
+when using ResourceClaimTemplate) during Pod creation and updates and
+optionally deletes it after Pod termination.
+
+Once a ResourceClaim is allocated and the Pod runs, there will be no periodic
+API calls. How much this impacts performance of the apiserver therefore mostly
+depends on how often this feature is used for new ResourceClaims and
+Pods. Because it is meant for long-running applications, the impact should not
+be too high.
 
 ###### Will enabling / using this feature result in introducing new API types?
 
-<!--
-Describe them, providing:
-  - API type
-  - Supported number of objects per cluster
-  - Supported number of objects per namespace (for namespace-scoped objects)
--->
+For DeviceClass, only a few (something like 10 to 20)
+objects per cluster are expected. Admins need to create those.
+
+The number of ResourceClaim objects depends on how much the feature is
+used. They are namespaced and get created directly or indirectly by users. In
+the most extreme case, there will be one or more ResourceClaim for each Pod.
+But that seems unlikely for the intended use cases.
+
+How many ResourcSlice objects get published depends on third-party drivers and
+how much hardware is installed in the cluster. Typically, each driver will
+publish one ResourceSlice per node where it manages hardware.
+
+Kubernetes itself will not impose specific limitations for the number of these
```

Comment:
Any etcd scale limits that we need to worry about, especially, if the ResourceSlices do get bigger with partionable devices support?

Reply:
I'm not aware of any besides the usual maximum size of a single object. The limits imposed on slices and string lengths must ensure that the object can be stored when encoded, which is why we did those back-of-the-envelope calculations about the maximum size.